### PR TITLE
fix(perps): batch bug fixes for trading form, token selector, and favorites(OK-51815 OK-51816 OK-51743 OK-51741 OK-50115)

### PR DESCRIPTION
**Pull request by Minnzen**
## Summary
- Fix trigger order not clearing button display state after submission
- Add cursor:pointer to Market/Limit/Trigger order type buttons on desktop hover
- Fix star icon hover shadow clipping in token selector rows
- Fix mobile token selector "Perps" tab showing blank and items disappearing on sort
- Fix Android favorites tab flashing "add favorites" empty state on every open
- Fix desktop token selector not scrolling to top on tab switch

## Intent & Context
Batch of Perps UI bug fixes reported by QA during App-6.1.0 testing cycle. Each fix addresses a specific user-facing issue that degrades the trading experience.

## Root Cause

**OK-51815 (Form not cleared):** In `SizeInput`, when `resetTradingForm()` clears both `size` and `triggerPrice` simultaneously, the `referencePrice` changes (falls back from triggerPrice to midPrice). The price-change effect fires in the same render cycle as the value-change effect, but reads the OLD `usdAmount` from its closure (not yet committed by setState). It then recalculates a token value and writes it back to `formData.size` via `onChange`, undoing the reset.

**OK-51816 (Cursor):** Order type tab buttons had `cursor="default"` instead of `cursor="pointer"`.

**OK-51743 (Star shadow clipping):** The `FavoriteButton` was inside an `overflow="hidden"` container, clipping its hover shadow on the left side.

**OK-51741 (Blank tab / disappearing items):** FlashList's internal layout cache and recycling mechanism doesn't properly handle significant data changes (tab switching from small to large datasets, or data reordering from sort changes). This is a known FlashList limitation — it preserves scroll offset and layout state across data changes, which causes blank renders or recycled items to disappear. **Trade-off note:** The fix uses React `key` prop (including activeTab + sort config) to force FlashList rebuild on tab/sort changes. This sacrifices FlashList's recycling optimization for correctness. For ~300 items with simple rows, the rebuild cost is negligible compared to the alternative of visible rendering bugs. A FlashList patch was considered but rejected — `key` is the idiomatic React solution for resetting component state, and the FlashList behavior (preserving scroll position) is correct for other use cases like real-time price updates.

**OK-50115 (Favorites flash):** `globalAtom({ persist: true })` initializes with `{ favorites: [] }` and loads persisted data asynchronously (`getOnInit = false` in jotaiStorage). On first render, favorites is empty, triggering `FavoritesEmptyState`. Android's slower storage makes the flash visible.

## Changes Detail
- `SizeInput.tsx`: Guard price-change effect with `prevValueRef.current` check — skip recalculation when value was externally cleared
- `PerpTradingForm.tsx`: Change `cursor="default"` to `cursor="pointer"` on Market/Limit tabs, add `cursor="pointer"` to Trigger dropdown
- `OrderTypeSelector.tsx`: Change `cursor="default"` to `cursor="pointer"` on mobile order type selector
- `PerpTokenSelectorRow.tsx`: Move `FavoriteButton` outside the `overflow="hidden"` container
- `MoblieTokenSelector.tsx`: Add composite `key` (tab + sort config) to FlashList, remove stale `scrollToOffset`, defer empty state until `isFavoritesReady`
- `PerpTokenSelector.tsx`: Add `key={activeTab}` to desktop FlashList, remove stale `scrollToOffset` from `setActiveTab`, defer empty state until `isFavoritesReady`
- `usePerpsFavorites.ts`: Expose `isReady` flag based on universe data load status

## Risk Assessment
- **Risk Level**: Low
- **Affected Platforms**: Desktop (OK-51816, OK-51743), Mobile (OK-51741, OK-50115), All (OK-51815)
- **Risk Areas**: FlashList key rebuild may cause brief re-render on tab/sort change; SizeInput guard could theoretically skip a legitimate price recalculation when the user manually clears input and price changes simultaneously (extremely unlikely edge case)

## Test plan
- [ ] Desktop: Hover Market/Limit/Trigger buttons — cursor should be pointer
- [ ] Desktop: Star icon hover shadow in token selector should not be clipped
- [ ] Desktop: Switch token selector tabs — list should start from top
- [ ] Mobile: Place a trigger order in USD mode — buttons should clear after submission
- [ ] Mobile: Switch between token selector tabs including "Perps" — no blank tab
- [ ] Mobile: Repeatedly sort within a small tab — no items should disappear
- [ ] Android: Open Perps with favorites tab active — should not flash "add favorites" page

---

<a href="https://app.devin.ai/review/onekeyhq/app-monorepo/pull/10702" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open with Devin
  </picture>
</a>

### CODE REVIEW

**Comment by revan-zhang** [comment]
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scan Engine |  Critical |  High |  Medium |  Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/d53f65b6-dede-488a-9edc-b393de614bdf?refs=5b0ab1d52d7ded9b0e13690a7d86c43be268ced3&source=prChecksComment) |
| :white_check_mark: | Licenses| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/d53f65b6-dede-488a-9edc-b393de614bdf/license?refs=5b0ab1d52d7ded9b0e13690a7d86c43be268ced3&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

File: packages/kit/src/views/Perp/hooks/usePerpsFavorites.ts [hunk 1]
```diff
@@ -66,5 +69,5 @@ export function usePerpsFavorites(): { favoriteItems: IFavoriteItem[] } {
     return items;
   }, [universe, favorites.favorites]);
 
-  return { favoriteItems };
+  return { favoriteItems, isReady: universe !== undefined };
```
[the remark this answers is not in the document]
Comment (Minnzen) [reply packages/kit/src/views/Perp/hooks/usePerpsFavorites.ts:72]: In practice, the persist atom loads from local AsyncStorage which resolves significantly faster than the universe API fetch. By the time `universe \!== undefined` becomes true, the favorites atom has already hydrated. A storage-ready flag would require infrastructure changes to the globalAtom system for a race condition that doesn't occur in real usage. The current heuristic is a pragmatic trade-off — verified on Android where the original bug was reported.

File: packages/kit/src/views/Perp/components/TokenSelector/MoblieTokenSelector.tsx
[the remark this answers is not in the document]
Comment (Minnzen) [reply packages/kit/src/views/Perp/components/TokenSelector/MoblieTokenSelector.tsx:1]: Intentional — showing a brief blank is preferred over flashing a misleading "No data" or "Add favorites" prompt. The loading window is very short (local storage hydration + universe fetch), and the list appears as soon as data is ready. Adding a loading spinner or placeholder for this sub-second transition adds unnecessary complexity.

File: packages/kit/src/views/Perp/components/TokenSelector/MoblieTokenSelector.tsx [hunk 2]
```diff
@@ -552,7 +550,7 @@ function MobileTokenSelectorModal({
             data={mockedListData}
             renderItem={renderItem}
             ListEmptyComponent={
-              activeTab === 'favorites' && !searchQuery ? (
+              activeTab === 'favorites' && !searchQuery && isFavoritesReady ? (
                 <FavoritesEmptyState isMobile />
               ) : (
```
[the remark this answers is not in the document]
Comment (Minnzen) [reply packages/kit/src/views/Perp/components/TokenSelector/MoblieTokenSelector.tsx:555]: Same as above — this is intentional. A brief blank during the sub-second hydration window is acceptable and preferred over adding a loading placeholder for this edge case.

**Review by originalix — APPROVED** [review]

**Review by zhaono1 — APPROVED** [review]
